CSP-1506: Decline Account Request - Confirmation Page

shaun-downey-education opened the pull request:
- Added a new shutter page to handle the decline scenario when reviewing an AddAccount request.

shaun-downey-education committed:
CSP-1611: WIP

shaun-downey-education committed:
CSP-1507: Your Training Providers - Manage Requests Functionality.

shaun-downey-education committed:
CSP-1507: SFA.DAS.Employer.PR.Domain.UnitTests project setup and additional code coverage.

shaun-downey-education committed:
CSP-1507: Additional test coverage for RequestsController.

shaun-downey-education committed:
CSP-1507: Sonar cloud code suggestions.

shaun-downey-education committed:
CSP-1507: Remove unnecessary null check on provider request model object.

shaun-downey-education committed:
CSP-1499: Intial review request permission page - WIP.

shaun-downey-education committed:
CSP-1507: Code review suggestions.

shaun-downey-education committed:
CSP-1507: Additional code review suggestion updates.

shaun-downey-education committed:
CSP-1507: Update naming convention on variable to match updated class name.

shaun-downey-education committed:
CSP-1499: WIP

shaun-downey-education committed:
CSP-1499: WIP

shaun-downey-education committed:
CSP-1507: Added request type to request response object to map correct view request urls to the training providers permissions table.

shaun-downey-education committed:
CSP-1499: Merged 1507

shaun-downey-education committed:
CSP-1499: Update review permission links.

shaun-downey-education committed:
CSP-1507: Additional unit test coverage.

shaun-downey-education committed:
CSP-1507: Remove constant file from coverage.

shaun-downey-education committed:
Merge branch 'CSP-1507_Your_training_providers_manage_requests' into CSP-1499-Review_permissions_request

shaun-downey-education committed:
CSP-1499: Re-validate request on submitting of review permissions page.

shaun-downey-education committed:
CSP-1507: Increate GetPermissionRequestResponse test coverage.

shaun-downey-education committed:
Merge branch 'CSP-1507_Your_training_providers_manage_requests' into CSP-1499-Review_permissions_request

shaun-downey-education committed:
CSP-1499: Update Permissions Request - Accept / Decline review permissions request with associated unit testing.

shaun-downey-education committed:
CSP-1499: ReviewPermissionsRequestSubmitViewModelValidator unit test coverage.

shaun-downey-education committed:
CSP-1499: Additional unit test coverage to handle the success banner properties on the your training providers page.

shaun-downey-education committed:
CSP-1499: Extract ternary statement to increase code readabiliity.

shaun-downey-education committed:
CSP-1499: Sonar cloud code suggestions  - remove deprecated scope and improve unit test names.

shaun-downey-education committed:
CSP-1499: Fix response null route bug by redirecting to correct action on the error controller.

shaun-downey-education committed:
CSP-1504: WIP

shaun-downey-education committed:
CSP-1499: Remove unnecessary response object from outer api endpoints.

shaun-downey-education committed:
CSP-1504: WIP

shaun-downey-education committed:
CSP-1499: Update Permissions - Set route parameter to employerAccountId to fall in line with authorization middleware.

shaun-downey-education committed:
CSP-1507: Centralize permission reference phrases, added AddAccountController test cases.

shaun-downey-education committed:
CSP-1504: Added additional unit test coverage.

shaun-downey-education committed:
CSP-1504: AddAccountController - Apply expected authorization policy.

shaun-downey-education committed:
CSP-1505: WIP

shaun-downey-education committed:
CSP-1504: Additional Test Coverage.

shaun-downey-education committed:
CSP-1504: Sonarcloud code smells.

shaun-downey-education committed:
Merge branch 'CSP-1504_Review_add_account_and_set_permissions_request' into CSP-1505_Decline_add_and_permission_request

shaun-downey-education committed:
CSP-1505: Decline add account request - Shutter Page.

shaun-downey-education committed:
CSP-1506: WIP

shaun-downey-education committed:
CSP-1505: Additional unit test coverage for AddAccountsController.

Moved radio button logic in the review add account request page into the view model.

shaun-downey-education committed:
CSP-1505: Remove unused setter.

shaun-downey-education committed:
Merge branch 'CSP-1505_Decline_add_and_permission_request' into CSP-1506_Conformation_decline_add_and_permissions_request

shaun-downey-education committed:
CSP-1506: Confirm Decline Add Account Request

shaun-downey-education committed:
CSP-1506: Update apprenticeship service help desk link.

DevenKShah commented on src/SFA.DAS.Employer.PR.Web/Controllers/YourTrainingProvidersController.cs:
```diff
@@ -119,12 +124,11 @@ private void SetSuccessBanner(YourTrainingProvidersViewModel model)
-
-        RequestAction requestAction = (RequestAction)Enum.Parse(typeof(RequestAction), requestActionTempValue);
+  
+        RequestAction requestActionEnum = (RequestAction)Enum.Parse(typeof(RequestAction), requestAction);
```
There is a generic `Enum.Parse<RequestAction>(requestAction)`

DevenKShah commented on src/SFA.DAS.Employer.PR.Web/Controllers/DeclineAddAccountConfirmationController.cs:
```diff
@@ -0,0 +1,40 @@
+{
+    public async Task<IActionResult> Index([FromRoute] Guid requestId, [FromRoute] string employerAccountId, CancellationToken cancellationToken)
+    {
+        var requestResponse = await _outerApiClient.GetRequest(requestId, cancellationToken);
+
+        if (!ReviewRequestHelper.IsValidRequest(requestResponse, RequestType.AddAccount))
+        {
+            return RedirectToRoute(RouteNames.YourTrainingProviders, new { employerAccountId });
+        }
+
+        await _outerApiClient.DeclineRequest(
+            requestId, 
+            new Domain.OuterApi.Permissions.DeclineRequestModel(User.GetUserId().ToString()),
+            cancellationToken
+        );
```
This should be done in the POST action of the previous page, this should be simply showing the view

DevenKShah reviewed (commented):
The changesaction should be carried out in the POST action instead of at redirection

shaun-downey-education committed:
CSP-1516: Update enum casting to use generic extension.

shaun-downey-education committed:
CSP-1506: Update initial request declined logic to allow the declining of the request when selecting the 'Decline request' button.

Added required unit test coverage.

shaun-downey-education committed:
CSP-1506: Fix Merge.

shaun-downey-education committed:
CSP-1506: Fix Merge.

shaun-downey-education committed:
CSP1506: Increase unit test coverage.

shaun-downey-education committed:
CSP-1506: Updated error messages on both review update permissions and review add account.

Additional text updates.

shaun-downey-education committed:
CSP-1506: Incorporate logic to handle requests without a link to an existing provider.

shaun-downey-education committed:
CSP-1506: Additional unit test coverage.

shaun-downey-education committed:
CSP-1506: Sonarcloud code suggestions.

shaun-downey-education committed:
CSP-1506: Merge master

shaun-downey-education committed:
CSP-1506: Merge master.

shaun-downey-education committed:
CSP-1506: Update view your training providers link to manage training providers.

shaun-downey-education committed:
CSP-1506: Update view your training providers link on the review add account request page.

shaun-downey-education committed:
CSP-1506:

- Various content updates.
- Remove pre-setting of accept / decline add account request when clicking the back button from the confirm decline request page.
- Fixed an issue where the error message on the review add account page does not take you to the radio button section when clicked.
- Updated unit tests to reflect changes.
- On both add account and update permissions, if the request is not valid, redirect to the cannot view request shutter page.

shaun-downey-education committed:
CSP-1506:

- Remove radio button associated parameters from DeclineRequest route.
- Updated associated unit tests.
